gammapy/estimators/map/ts.py

```diff
@@ -236,10 +237,13 @@ def estimate_kernel(self, dataset):
         model = self.model.copy()
         model.spatial_model.position = geom.center_skydir
 
-        # Creating exposure map with exposure at map center
+        # Creating exposure map with the mean non-null exposure
         exposure = Map.from_geom(geom, unit="cm2 s1")
-        exposure_center = dataset.exposure.to_region_nd_map(geom.center_skydir)
-        exposure.data[...] = exposure_center.data
+        position = get_nearest_valid_exposure_position(dataset.exposure, geom.center_skydir)
+        exposure_position = dataset.exposure.to_region_nd_map(position)
+        if not np.any(exposure_position.data):
+            raise ValueError("The exposure is null. Impossible to convolve the model with the PSFKernel.")
```

Comment:
I would use a simpler error message e.g.:
`No valid exposure. Impossible to compute kernel for TS Map.`